addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -1759,3 +1759,34 @@ def test_multi_users_enabled(self):
         self.assertEqual(res_pick.move_line_ids, expected_backorder_mls)
         # Original picking in state done
         self.assertEqual(pick.state, "done")
+
+
+class TestStockPickingPriorities(common.BaseUDES):
+    def test_priority_remains_on_done_picking(self):
+        """
+        Test that priorities are not overwritten on done pickings.
+        """
+        self.create_quant(self.fig.id, self.test_stock_location_02.id, 100)
```

Comment:
You've got a lot of repeated code here, as you basically have the same picking it's just that in one test you complete it and the other you cancel.

Let's move the code that creates the quant and picking into `setUpClass`, that will cut down the tests a fair bit.